src/power_grid_model_ds/_core/load_flow.py

```diff
@@ -63,6 +63,23 @@ def create_input_from_grid(self):
             self.input_data[array_name] = pgm_array
         return self.input_data
 
+    def create_grid_from_input_data(
+        self,
+    ) -> Grid:
+        """
+        Create Grid object from PowerGridModel input.
+        Note that for some arrays, not all fields are available in the PowerGridModel input.
+        In this case, the default values are used.
+
+        Returns a Grid object with the arrays filled with the PowerGridModel input.
+        """
+        for pgm_name in PGM_ARRAYS:
+            if pgm_name in self.input_data:
+                pgm_ds_array_class = getattr(self.grid, pgm_name).__class__
+                pgm_ds_array = pgm_ds_array_class(self.input_data[pgm_name])
+                self.grid.append(pgm_ds_array, check_max_id=False)
+        return self.grid
```

Comment:
```suggestion
    def create_grid_from_input_data(self, check_ids: bool = True) -> Grid:
        """
        Create Grid object from PowerGridModel input.
        Note that for some arrays, not all fields are available in the PowerGridModel input.
        In this case, the default values are used.

        Args:
            check_ids: if True, check if the ids are unique

        Returns a Grid object with the arrays filled with the PowerGridModel input.
        """
        for pgm_name in PGM_ARRAYS:
            if pgm_name in self.input_data:
                pgm_ds_array_class = getattr(self.grid, pgm_name).__class__
                pgm_ds_array = pgm_ds_array_class(self.input_data[pgm_name])
                self.grid.append(pgm_ds_array, check_max_id=False)
        if check_ids:
            self.grid.check_ids()
        return self.grid
```

I suggest adding a check on ids option, since the grid object is not necessarily empty. This I think makes it safe enough

Reply:
Just to note: this way the grid will already have duplicated ids by the time check_ids() raises an error.

I still think it is fine to go with this solution

Reply:
It is correct that in this setup the check will raise an error while having created an 'invalid grid', I think this is fine since it is not an expected path which needs recovering from.

On the alternative, that will block more than just non-unique ids since arrays are added one-by-one and no order in ids is expected. So if the node id has 2 and the line id has 1 that would raise an error while the ids are fine.